python/pyarrow/tests/test_array.py

```diff
@@ -991,6 +991,16 @@ def test_map_labelled():
     assert len(arr) == 2
 
 
+def test_map_from_dict():
+    # ARROW-17832
+    tup_arr = pa.array([[('a', 1), ('b', 2)], [('c', 3)]],
+                       pa.map_(pa.string(), pa.int64()))
+    dict_arr = pa.array([{'a': 1, 'b': 2}, {'c': 3}],
```

Comment:
Thanks for this fix. Just had a quick followup question though. Should the nested dict work too? e.g.)

```python
import pyarrow as pa
pa.array([[{'a': 1, 'b': 2}, {'c': 3}]], pa.list_(pa.map_(pa.string(), pa.int64())))
```

It currently fails w/

```
pyarrow.lib.ArrowTypeError: Could not convert {'a': 1, 'b': 2} with type dict: was not a sequence or recognized null for conversion to list type
```

Reply:
I would assume the nested case to work as well. Can you open a new issue about this?

Reply:
Thanks guys. I filed an issue here: https://github.com/apache/arrow/issues/33928